src/main/scala/Engine.scala

```diff
@@ -36,12 +37,17 @@ case class Query(
     fields: Option[List[Field]] = None, // default: whatever is in algorithm params or None
     currentDate: Option[String] = None, // if used will override dateRange filter, currentDate must lie between the item's
     // expireDateName value and availableDateName value, all are ISO 8601 dates
-    dateRange: Option[DateRange] = None, // optional before and after filter applied to a date field
-    blacklistItems: Option[List[String]] = None, // default: whatever is in algorithm params or None
+    dateRange: Option[DateRange] = None, // default: whatever is in algorithm params or None
     returnSelf: Option[Boolean] = None,// means for an item query should the item itself be returned, defaults
                                        // to what is in the algorithm params or false
     num: Option[Int] = None, // default: whatever is in algorithm params, which itself has a default--probably 20
-    eventNames: Option[List[String]]) // names used to ID all user actions
+    eventNames: Option[List[String]], // names used to ID all user actions
+    /** following variable defines features - must include particular items, price range, blacklist items,blacklistCategories respectively ***/
+    includeItems: Option[List[String]] = None,
+    priceRange: Option[PriceRange] = None,// optional before and after filter applied to a date field
+    blacklistItems: Option[List[String]] = None,
+    blacklistCategory: Option[List[String]] = None,
+    blacklistBrand: Option[List[String]] = None)
   extends Serializable
```

Comment:
Hi Pat

Just read your suggestions. Will work on it.
Once completed will send you PR